File: ios/Classes/Player/Models/AdvertisementRequest.swift
//
//  AdvertisementRequest.swift
//  udevs_video_player
//
//  Created by Abdurahmon on 26/07/2024.
//

import Foundation


struct AdvertisementRequest{
    
    var device: String = "mobile"
    var format: String = "player"
    var age: Int
    var gender:String
    var paymentType: String
    var region: String
    var userId: String
    
    
    func toMap()->[String:Any]{
        return [
            "device": device,
            "format":format,
            "age":age,
            "gender":gender,
            "payment_type":paymentType,
            "region":region,
            "user_id":userId
        ]
    }
}

File: android/src/main/kotlin/uz/udevs/udevs_video_player/models/advertisement/AdvertisementRequest.kt
package uz.udevs.udevs_video_player.models.advertisement

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class AdvertisementRequest(
    @SerializedName("device")
    val device: String = "mobile",
    @SerializedName("format")
    val format: String = "player",
    @SerializedName("age")
    val age: Int,
    @SerializedName("gender")
    val gender: String,
    @SerializedName("payment_type")
    val paymentType: String,
    @SerializedName("region")
    val region: String,
    @SerializedName("user_id")
    val userId: String,
) : Serializable